controllers/helmrelease_controller.go

```diff
@@ -678,11 +683,42 @@ func (r *HelmReleaseReconciler) reconcileDelete(ctx context.Context, hr v2.HelmR
 		if err != nil {
 			return ctrl.Result{}, err
 		}
-		if err := run.Uninstall(hr); err != nil && !errors.Is(err, driver.ErrReleaseNotFound) {
-			return ctrl.Result{}, err
-		}
-		ctrl.LoggerFrom(ctx).Info("uninstalled Helm release for deleted resource")
 
+		impersonator := runtimeClient.NewImpersonator(
+			r.Client,
+			r.StatusPoller,
+			r.PollingOpts,
+			hr.Spec.KubeConfig,
+			r.KubeConfigOpts,
+			kube.DefaultServiceAccountName,
+			hr.Spec.ServiceAccountName,
+			hr.GetNamespace(),
+		)
+		if impersonator.CanImpersonate(ctx) {
+			if err := run.Uninstall(hr); err != nil && !errors.Is(err, driver.ErrReleaseNotFound) {
+				return ctrl.Result{}, err
+			}
+			log.Info("uninstalled Helm release for deleted resource")
+		} else {
+			release, err := run.ObserveLastRelease(hr)
```

Comment:
Did you test this? I would expect this to not be able to reach the Secret, as the `getter` configured makes use of impersonation as well.

Reply:
yes i have and it works fine. the getter does make use of the impersonation, but it doesn't actually need the ServiceAccount to exist as long as there exists a RoleBinding which points to a Role with sufficient perms.

Reply:
To test this, create a HR in a namespace along with the SA and the role binding, delete the namespace and see it stuck in termination.